x-pack/plugins/data_enhanced/server/search/session/session_service.ts

```diff
@@ -406,17 +388,15 @@ export class SearchSessionService implements ISessionService {
       });
       const deps = { savedObjectsClient };
       return {
-        search: <Request extends IKibanaSearchRequest, Response extends IKibanaSearchResponse>(
-          strategy: ISearchStrategy<Request, Response>,
-          ...args: Parameters<ISearchStrategy<Request, Response>['search']>
-        ) => this.search(strategy, ...args, deps),
-        save: (sessionId: string, attributes: Partial<SearchSessionSavedObjectAttributes>) =>
-          this.save(sessionId, attributes, deps),
-        get: (sessionId: string) => this.get(sessionId, deps),
-        find: (options: SavedObjectsFindOptions) => this.find(options, deps),
-        update: (sessionId: string, attributes: Partial<SearchSessionSavedObjectAttributes>) =>
-          this.update(sessionId, attributes, deps),
-        delete: (sessionId: string) => this.delete(sessionId, deps),
+        getId: this.getId.bind(this, deps),
+        trackId: this.trackId.bind(this, deps),
+        getSearchIdMapping: this.getSearchIdMapping.bind(this, deps),
+        save: this.save.bind(this, deps),
+        get: this.get.bind(this, deps),
+        find: this.find.bind(this, deps),
+        update: this.update.bind(this, deps),
+        extend: this.extend.bind(this, deps),
+        cancel: this.cancel.bind(this, deps),
```

Comment:
@lukasolson could you please keep both `cancel` and `delete` APIs, so we have some flexibility?

Reply:
https://github.com/elastic/kibana/pull/90015